File: module_2/multi_lang/email_val.js
// Prompt: "Function to validate email format and check if user exists in database using Mongoose"
// Copilot will suggest:
// - regex pattern for validation
// - async/await for database query
// - Promise-based error handling
const User = require('./models/User');

async function validateAndCheckEmail(email) {
    // Regular expression for basic email validation
    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;

    // Validate email format
    if (!emailRegex.test(email)) {
        throw new Error('Invalid email format');
    }

    // Check if user exists in the database
    const user = await User.findOne({ email: email });
    if (!user) {
        throw new Error('User does not exist');
    }
    return true; // Email is valid and user exists
}

File: module_2/multi_lang/email_val.py
# Prompt: "Function to validate email format and check if exists in database using SQLAlchemy"
# Copilot will suggest:
# - regex pattern matching for email validation
# - SQLAlchemy query to check database
# - proper error handling
import re
from models import User # type: ignore

def validate_and_check_email(email):
    """Validate email format and check if exists in database."""
    # Regex pattern for validating an email
    email_pattern = r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$'
    
    # Validate email format
    if not re.match(email_pattern, email):
        return False, "Invalid email format."
    
    # Check if email exists in the database
    user = User.query.filter_by(email=email).first()
    if user:
        return True, "Email exists in the database."
    else:
        return False, "Email does not exist in the database."
